quinn/src/builders.rs

```diff
@@ -30,12 +30,11 @@ where
     S: proto::crypto::Session + Send + 'static,
 {
     /// Start a builder with a specific initial low-level configuration.
-    pub fn new(config: EndpointConfig<S>) -> Self {
+    pub fn new(config: EndpointConfig<S>, default_client_config: ClientConfig<S>) -> Self {
```

Comment:
I don't like this part that much. Could we make the field `None` initially? I suppose then we have to pay for per-`Endpoint` construction of the `ClientConfig` at `build()` time if the user doesn't set one explicitly.

Reply:
I feel like this is consistent with the existing use of `new`, which is a niche alternative to `default()` when you already have an `EndpointConfig` for some reason. Sharing instances of the expensive-to-construct default `ClientConfig` between cloned builders seems desirable as well.

Reply:
Fair enough!